### PR TITLE
Added handling for query failures

### PR DESCRIPTION
**Pull request by emmberk**

### CODE REVIEW

File: multi-int-store/src/main/java/com/connexta/multiintstore/controllers/QueryController.java [hunk 1]
```diff
@@ -34,12 +34,16 @@
   @RequestMapping(method = RequestMethod.GET, params = "q")
   @ResponseBody
   public ResponseEntity<List<URL>> searchKeyword(@RequestParam(value = "q") String keyword) {
+    if (StringUtils.isEmpty(keyword)) {
```
Comment (ahoffer) [inline multi-int-store/src/main/java/com/connexta/multiintstore/controllers/QueryController.java]: We might be able use the Bean Validation framework's annotation based validators.
I'm not immediately sure how. Here is a link to a Baeldung article that discusses them.
https://www.baeldung.com/spring-boot-bean-validation

**Comment by emmberk** [comment]
rebased

**Comment by emmberk** [comment]
rebased

File: multi-int-store/src/main/java/com/connexta/multiintstore/controllers/QueryController.java (hunk 1 above)
Comment (emmberk) [reply multi-int-store/src/main/java/com/connexta/multiintstore/controllers/QueryController.java]: Good idea. I just pushed this update:
```java
@GetMapping
@ResponseBody
public ResponseEntity<List<URL>> searchKeyword(
    @RequestParam(value = "q") @NotEmpty final String keyword) {
  try {
    return new ResponseEntity<>(searchService.find(keyword), HttpStatus.OK);
  } catch (SearchException e) {
    LOGGER.warn("Unable to search for {}", keyword, e);
  }

  return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
}

@ControllerAdvice
private class ConstraintViolationExceptionHandler extends ResponseEntityExceptionHandler {

  @ExceptionHandler(value = {ConstraintViolationException.class})
  protected ResponseEntity<Object> handleConstraintViolation(
      @NotNull final ConstraintViolationException e, @NotNull final WebRequest request) {
    final String message = e.getMessage();
    final HttpStatus httpStatus = HttpStatus.BAD_REQUEST;
    LOGGER.warn("Request is invalid: {}. Returning {}.", message, httpStatus, e);
    return handleExceptionInternal(e, message, new HttpHeaders(), httpStatus, request);
  }
}
```

**Comment by emmberk** [comment]
build now

**Pull request merged**